sources/packages/api/src/route-controllers/esdc-integration/e-cert-integration.system.controller.ts

```diff
@@ -3,37 +3,67 @@ import { InjectLogger } from "../../common";
 import { LoggerService } from "../../logger/logger.service";
 import { AllowAuthorizedParty } from "../../auth/decorators";
 import { AuthorizedParties } from "../../auth/authorized-parties.enum";
-import { ECertFullTimeRequestService } from "../../esdc-integration/e-cert-full-time-integration/e-cert-full-time-request.service";
-import { ECertFullTimeResponseService } from "../../esdc-integration/e-cert-full-time-integration/e-cert-full-time-response.service";
-import { ESDCFileResultDTO, ESDCFileResponseDTO } from "./models/esdc-model";
+import { ECertFileHandler } from "../../esdc-integration/e-cert-integration/e-cert-file-handler";
+import { ESDCFileResponseDTO, ESDCFileResultDTO } from "./models/esdc-model";
 import { ApiTags } from "@nestjs/swagger";
 import BaseController from "../BaseController";
+import { OfferingIntensity } from "../../database/entities";
+import { ECertFullTimeResponseService } from "../../esdc-integration/e-cert-integration/e-cert-full-time-integration/e-cert-full-time-response.service";
 
+const ECERT_FULL_TIME_FILE_CODE = "PBC.EDU.ECERTS.D";
+const ECERT_PART_TIME_FILE_CODE = "PBC.EDU.PTCERTS.D";
+const ECERT_FULL_TIME_SENT_FILE_SEQUENCE_GROUP = "ECERT_FT_SENT_FILE";
+const ECERT_PART_TIME_SENT_FILE_SEQUENCE_GROUP = "ECERT_PT_SENT_FILE";
 @AllowAuthorizedParty(AuthorizedParties.formsFlowBPM)
 @Controller("system-access/e-cert")
 @ApiTags("system-access")
 export class ECertIntegrationController extends BaseController {
   constructor(
-    private readonly ecertFullTimeRequestService: ECertFullTimeRequestService,
+    private readonly eCertFileHandler: ECertFileHandler,
     private readonly eCertFullTimeResponseService: ECertFullTimeResponseService,
   ) {
     super();
   }
 
   /**
    * Process disbursements available to be sent to ESDC.
-   * Considerer any record that is scheduled in upcoming days or in the past.
+   * Consider any record that is scheduled in upcoming days or in the past.
    * @returns result of the file upload with the file generated and the
    * amount of records added to the file.
    */
-  @Post("process")
+  @Post("process-full-time")
   async processFullTimeECertFile(): Promise<ESDCFileResultDTO> {
-    this.logger.log("Sending full time e-Cert File...");
-    const uploadResult = await this.ecertFullTimeRequestService.generateECert();
-    this.logger.log("Full time e-Cert file sent.");
+    this.logger.log("Sending Full-Time E-Cert File...");
+    const uploadFullTimeResult = await this.eCertFileHandler.generateECert(
```

Comment:
Can we maybe have one specific method for each FT/PT and internally deal with these parameters since they will be always the same for FT and always the same for PT.
The method would be
```typescript
const uploadFullTimeResult = await this.eCertFileHandler.generateFullTimeECert();
```